src/pages/ProjectSettings.tsx

```diff
@@ -34,13 +37,30 @@ export const ProjectSettings: React.FC = () => {
     tab: ProjectSettingsTabRoutes;
   }>();
 
-  const { data } = useQuery<
+  // If the path includes an Object ID, this page represents a repo and we should not attempt to fetch a project.
+  const isRepo = validateObjectId(identifier);
+
+  const { data: projectData = null } = useQuery<
     ProjectSettingsQuery,
     ProjectSettingsQueryVariables
   >(GET_PROJECT_SETTINGS, {
+    skip: isRepo,
     variables: { identifier },
   });
 
+  const repoRefId =
+    projectData?.projectSettings?.projectRef?.repoRefId || identifier;
+  const useRepoSettings =
+    projectData?.projectSettings?.projectRef?.useRepoSettings;
+
+  const { data: repoData } = useQuery<
+    RepoSettingsQuery,
+    RepoSettingsQueryVariables
+  >(GET_REPO_SETTINGS, {
+    skip: useRepoSettings === false,
```

Comment:
does [useLazyQuery](https://www.apollographql.com/docs/react/data/queries/#manual-execution-with-uselazyquery) work here instead?

Reply:
I think this query will always be dispatched since the initial render value of useRepoSettings will be undefined

Reply:
adjusting line 53 to  `const useRepoSettings = projectData?.projectSettings?.projectRef?.useRepoSettings ?? false` could prevent this